doc/hcr.rst

```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
 .. code-block:: nim
 
   # logic.nim
-  import sdl2/sdl
-
+  import sdl2
+  
   #*** import the hotcodereloading stdlib module ***
   import hotcodereloading
-
+  
   var runGame*: bool = true
-  var window: Window
-  var renderer: Renderer
-
+  var window: WindowPtr
+  var renderer: RendererPtr
+  var evt = sdl2.defaultEvent
+  
   proc init*() =
-    discard sdl.init(INIT_EVERYTHING)
-    window = createWindow("testing", WINDOWPOS_UNDEFINED.cint, WINDOWPOS_UNDEFINED.cint, 640, 480, 0'u32)
-    assert(window != nil, $sdl.getError())
+    discard sdl2.init(INIT_EVERYTHING)
+    window = createWindow("testing", SDL_WINDOWPOS_UNDEFINED.cint, SDL_WINDOWPOS_UNDEFINED.cint, 640, 480, 0'u32)
+    assert(window != nil, $sdl2.getError())
     renderer = createRenderer(window, -1, RENDERER_SOFTWARE)
-    assert(renderer != nil, $sdl.getError())
-
+    assert(renderer != nil, $sdl2.getError())
+  
   proc destroy*() =
     destroyRenderer(renderer)
     destroyWindow(window)
-
-  var posX = 1
-  var posY = 0
-  var dX = 1
-  var dY = 1
-
+  
+  var posX : cint = 1
+  var posY : cint = 0
+  var dX : cint = 1
+  var dY : cint = 1
+  
   proc update*() =
-    for evt in events():
-      if evt.kind == QUIT:
+    while pollEvent(evt):
+      if evt.kind == QuitEvent:
         runGame = false
         break
-      if evt.kind == KEY_DOWN:
-        if evt.key.keysym.scancode == SCANCODE_ESCAPE: runGame = false
-        elif evt.key.keysym.scancode == SCANCODE_F9:
+      if evt.kind == KeyDown:
+        if evt.key.keysym.scancode == getScancodeFromName("Escape"): runGame = false
+        elif evt.key.keysym.scancode == getScancodeFromName("F9"):
           #*** reload this logic.nim module on the F9 keypress ***
           performCodeReload()
-
+  
     # draw a bouncing rectangle:
     posX += dX
     posY += dY
-
+  
     if posX >= 640: dX = -2
     if posX <= 0: dX = +2
     if posY >= 480: dY = -2
     if posY <= 0: dY = +2
-
-    discard renderer.setRenderDrawColor(0, 0, 255, 255)
-    discard renderer.renderClear()
-    discard renderer.setRenderDrawColor(255, 128, 128, 0)
-
-    var rect: Rect(x: posX - 25, y: posY - 25, w: 50, h: 50)
-    discard renderer.renderFillRect(rect.addr)
+  
+    discard renderer.setDrawColor(0, 0, 255, 255)
+    discard renderer.clear()
+    discard renderer.setDrawColor(255, 128, 128, 0)
+  
+    var rect: Rect
+    rect = (x: posX - 25, y: posY - 25, w: 50.cint, h: 50.cint)
+    discard renderer.fillRect(rect.addr)
```

Comment:
```suggestion
    discard renderer.fillRect(rect)
```

There's a variant that takes a `var Rect`.

Reply:
Cheers alaviss
commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` resolved this